FinancialTracker/Firebase.swift

```diff
@@ -66,20 +72,49 @@ class FirebaseHandler {
     }
 
     func signIn(email: String, password: String, completionHandler: @escaping (FirebaseError?, Bool) -> Void) {
-        Auth.auth().signIn(withEmail: email, password: password) { _, error in
+        auth.signIn(withEmail: email, password: password) { _, error in
 
-            if error != nil {
+            guard error == nil else {
                 completionHandler(FirebaseError.auth(error), false)
+                return
             }
             completionHandler(nil, true)
         }
     }
     
-    func signOut() {
+    func signOut(completionHandler: @escaping (FirebaseError?, Bool) -> Void) {
         do {
-            try Auth.auth().signOut()
+            try auth.signOut()
         } catch let signOutError {
-            print("Error signing out: %@", FirebaseError.signOut(signOutError))
+            completionHandler(FirebaseError.signOut(signOutError), false)
+        }
+    }
+    
+    func getCurrentUserData(completionHandler: @escaping (FirebaseError?, Bool, [String: Any]?) -> Void) {
+        guard let uid = auth.currentUser?.uid else {
+            fatalError("Can't access without logged in user.")
```

Comment:
This probably should not be a `fatalError`. 
A prompt for the user to log in will work better. 

If you plan to implement that prompt later on, add a descriptive message or a`"#warning`